src/commands/unik/disclose.ts

```diff
@@ -107,20 +115,45 @@ export class UnikDiscloseCommand extends WriteCommand {
         return await this.formatResult(transactionFromNetwork, transactionStruct.id as string);
     }
 
-    private async checkExplicitValues(unikId: string, unikType: DIDType, listOfExplicitValues: string[]) {
+    private async checkExplicitValues(
+        unikId: string,
+        unikType: DIDType,
+        listOfExplicitValues: string[],
+    ): Promise<void> {
         // Check explicit values
+        let hasTDLExplicitValue: boolean = false;
         for (const explicit of listOfExplicitValues) {
             try {
-                const fingerPrintResult = await this.unsClientWrapper.computeTokenId(explicit, unikType, "UNIK");
-                if (fingerPrintResult !== unikId) {
+                const fingerPrintResult: FingerprintResult = await this.unsClientWrapper.computeTokenId(
+                    explicit,
+                    unikType,
+                    "UNIK",
+                    true,
+                );
+                if (fingerPrintResult.fingerprint !== unikId) {
                     throw new Error("At least one expliciteValue is not valid for this unikid");
                 }
+                if (fingerPrintResult.computingInformations?.isEndingWithTLDWithDot) {
+                    hasTDLExplicitValue = true;
+                }
             } catch (e) {
                 this.debug(`disclose-explicit-values - ${e.message}`);
                 throw new Error("At least one expliciteValue does not match expected format");
             }
         }
 
+        if (hasTDLExplicitValue && unikType !== "INDIVIDUAL") {
+            const userConsent: boolean = await this.warnUserAndGetConsent(
+                `${
+                    listOfExplicitValues.length > 1 ? "At least one @uniknameID of your list" : "This @uniknameID"
```

Comment:
why using @uniknameID and @unikname ? I don't see the point... but maybe Iam too stupid to understand the product ^^

Reply:
Spec, see with PO ;)

Reply:
je savais que t'allais dire ça !! tu donne pas ton avis ? Pr validée de mon coté.